www/assets/js/db-handler.js

```diff
@@ -174,6 +174,36 @@ var dbHandler = {
       if (!oldVersion)
         resolve();
 
+      if (oldVersion < 34) {
+        // Remove leading zero from barcodes and add originalBarcode property
+        await new Promise(function(resolve, reject) {
+          store = transaction.objectStore('foodList');
+
+          store.openCursor().onsuccess = function(event) {
+            let cursor = event.target.result;
+
+            if (cursor) {
+              let value = cursor.value;
+
+              if (value.barcode !== undefined) {
+                if (value.barcode.startsWith("0")) {
+                  value.originalBarcode = value.barcode;
+                  value.barcode = value.barcode.substring(1);
+                }
+                if (value.barcode.startsWith("fdcId_")) {
```

Comment:
I believe the "fdcId" prefix is to avoid clashes with barcodes from other systems such as open food facts - although it's been so long since it was written that I can't remember for certain. Do you think we'll run into any issues by removing the prefix?

Reply:
I only remove it for the `originalBarcode` property which is used for displaying and searching. The `barcode` property still has the prefix.

Reply:
Ah ok, gotcha! Time for a new release, or do you have some more commits to make first?

Reply:
In short, items from OFF will be stored like this:

```
barcode: "<barcode-without-leading-zero>"
originalBarcode: "<barcode-including-leading-zero>"
```

And items from USDA will be stored like this:

```
barcode: "fdcId_<id>"
originalBarcode: "<id>"
```

Reply:
> Time for a new release, or do you have some more commits to make first?

I'm done for the time being, you can go ahead with the release 🚀

Reply:
Thanks as always